croquemort/crawler.py

```diff
@@ -45,17 +51,21 @@ def check_url(self, url_group_frequency):
             if frequency:
                 self.storage.store_frequency(url, group, frequency)
         try:
-            try:
-                response = session.head(url, allow_redirects=True,
-                                        timeout=HEAD_TIMEOUT)
-            except requests.exceptions.ReadTimeout:
-                # simulate 404 to trigger GET request below
-                response = FakeResponse(status_code=404, headers={})
+            head_offend = any([re.match('^(http|https|ftp)://%s' % d, url)
```

Comment:
Maybe faster to extract the domain once with `urlsplit()` and test if it's in the list, no ?

Reply:
Maybe :-) I'll do that.